src/hotspot/cpu/s390/macroAssembler_s390.cpp

```diff
@@ -681,6 +681,17 @@ void MacroAssembler::add2reg(Register r1, int64_t imm, Register r2) {
   z_agfi(r1, imm);
 }
 
```

Comment:
I would prefer this method to have the same interface as add2reg(). Of course, that implies a more complex method body. Your choice.